server/container/guice/utils/src/main/java/org/apache/james/utils/ExtensionConfiguration.java

```diff
@@ -27,22 +27,31 @@
 import com.google.common.collect.ImmutableList;
 
 public class ExtensionConfiguration {
-    public static final ExtensionConfiguration DEFAULT = new ExtensionConfiguration(ImmutableList.of());
+    public static final ExtensionConfiguration DEFAULT = new ExtensionConfiguration(ImmutableList.of(), ImmutableList.of());
 
     public static ExtensionConfiguration from(Configuration configuration) {
         return new ExtensionConfiguration(
             Arrays.stream(configuration.getStringArray("guice.extension.module"))
+                .map(ClassName::new)
+                .collect(ImmutableList.toImmutableList()),
+            Arrays.stream(configuration.getStringArray("guice.extension.startable"))
                 .map(ClassName::new)
                 .collect(ImmutableList.toImmutableList()));
     }
 
     private final List<ClassName> additionalGuiceModulesForExtensions;
+    private final List<ClassName> startables;
 
-    public ExtensionConfiguration(List<ClassName> additionalGuiceModulesForExtensions) {
+    public ExtensionConfiguration(List<ClassName> additionalGuiceModulesForExtensions, List<ClassName> startables) {
```

Comment:
:thinking: it's hard to tell since I never used this mechanism before but looking @ org.apache.james.modules.server.ElasticSearchMetricReporterModule it looks like I can already declare 
```
guice.extension.module=org.apache.james.modules.server.ElasticSearchMetricReporterModule
```
or 
```
guice.extension.module=org.apache.james.modules.server.GraphiteMetricReporterModule
```
and that would be installed and the `start` would be called during server startup when InitializationOperation are collected. 

Therefore all the UserStartable extension is unneeded or am I missing something ?

Reply:
I was confused because in the existing apps the ElasticSearchMetricReporterModule is always explicitely installed in the main, is there a difference I am missing ?

Reply:
> won't it be picked up when loading all the InitializationOperation ?

No. 

Because InitializationOperation is used by core James stuff thus not open for use.

That also coupled with the way child injectors work in guice.

Reply:
alright thanks for the clarification